### PR TITLE
Error mixed table

### PR DESCRIPTION
**Pull request by Midnightific**
Closes: #572 

It makes the error message for mixed tables more descriptive and explains why they're bad and should be _avoided_.

### CODE REVIEW

File: selene-lib/src/lints/mixed_table.rs [hunk 1]
```diff
@@ -30,7 +30,7 @@ impl Lint for MixedTableLint {
         for mixed_table in visitor.mixed_tables {
             diagnostics.push(Diagnostic::new_complete(
                 "mixed_table",
-                "mixed tables are not allowed".to_owned(),
+                "mixed tables are heavily discouraged due to factors such as: type safety, readability, compatibility, and debugging".to_owned(),
```
Comment (Kampfkarren) [inline selene-lib/src/lints/mixed_table.rs:33]: The message here does not really adequately explain why they should be avoided. Also, "type safety" is not a thing outside of Luau. I don't think it's ever possible to do that in a small message, so I suggest something more openly vague:

```suggestion
                "mixed tables should be avoided, as they can cause confusing and hard to debug issues such as during iteration or encoding".to_owned(),
```

Comment (Midnightific) [reply selene-lib/src/lints/mixed_table.rs]: Aye aye! *Sorry I forgot that selene isn't a linter for just Roblox*

**Comment by Midnightific** [comment]
Scrap that review, forgot to change the stderr file, will do that tomorrow.

**Comment by Midnightific** [comment]
@Kampfkarren All good to review now!

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ## [Unreleased](https://github.com/Kampfkarren/selene/compare/0.26.1...HEAD)
 ### Added
 - Added `CFrame.lookAlong` to the Roblox standard library
+- When a mixed table is detected, selene will now tell you why mixed tables are bad and that they should be avoided
```
Comment (chriscerie) [inline CHANGELOG.md:7]:
```suggestion

### Changed
- Updated the warning message for the `must_use` lint to include why mixed tables should be avoided
```

**Pull request merged**